src/app/common/utils.ts

```diff
@@ -254,3 +254,18 @@ type WhenPageModeMap<T> = Record<PageMode, T>;
 export function whenPageMode<T>(pageModeMap: WhenPageModeMap<T>) {
   return pageModeMap[pageMode];
 }
+interface WhenNetworkMap<T> {
+  [DefaultNetworkIds.mainnet]: T;
+  [DefaultNetworkIds.testnet]: T;
+  [DefaultNetworkIds.devnet]: T;
+}
+export function whenNetwork(networkId: DefaultNetworkNames) {
+  return <T>(networkMap: WhenNetworkMap<T>): T => networkMap[networkId];
+}
+
+export function sumByKey<T extends PropertyKey>(
+  list: ({ [K in T]: number } & { [K in PropertyKey]?: unknown })[],
+  key: T
+) {
+  return list.reduce((sum, item) => sum.plus(item[key]), new BigNumber(0));
+}
```

Comment:
This is good, but do wonder if it's slightly over-complex / inflexible for our use case. For example, how do I sum an array here, as it expects an array of objects.

Isn't summing just an array easier, and using a map on the utxo array to get an array of numbers?